tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -9,7 +9,7 @@ load "${BATS_TEST_DIRNAME}/lib.sh"
 load "${BATS_TEST_DIRNAME}/confidential_common.sh"
 
 setup() {
-    confidential_setup && skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
+    confidential_setup_non_tees || skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
```

Comment:
This is maybe slightly outside of the scope of this PR, but it feels like the `confidential_setup_non_tees`/ `confidential_setup` logic is pretty confusing with various negation and requiring `||` vs. `&&`. This lead to the original mistake that at least three people overlooked.

This might be controversial, but I wonder if we would have better clarity (and avoid the issue we had here) with some renaming and separation and a bit more code. So this code would look something like:
```
if [ ! is_confidential_runtime_class ] then;
    skip "Test not supported for ${KATA_HYPERVISOR}."
fi
if [ is_confidential_hardware ]; then
     skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
fi
```
It isn't as succinct coding wise, but I'd argue that the best code is the easier to understand rather than the shortest.

We could create a confidential_common function that wraps this and was called something like `skip_for_non_confidential_and_tee_hardware` that would add clarity as well, to avoid the duplication of this code in the bats files?

What do others thinks?